src/plugins/data_source/common/data_sources/types.ts

```diff
@@ -11,8 +11,15 @@ export interface DataSourceAttributes extends SavedObjectAttributes {
   endpoint: string;
   auth: {
     type: AuthType;
-    credentials: UsernamePasswordTypedContent | undefined;
+    credentials: UsernamePasswordTypedContent | SigV4Content | undefined;
   };
+  lastUpdatedTime?: string;
+}
+
+export interface SigV4Content {
```

Comment:
I found the support of `aoss` was release 3 weeks ago in version `opensearch-js 2.2`. My initial dev was 2 month ago based on version `2.1.0`, where `service = es` is hardcoded in the client libarary. Also I haven't tested with serverless, given the 2.6 release is approaching. I propose we create an issue to track it, instead of adding to this PR

https://github.com/opensearch-project/opensearch-js/blob/2.1.0/lib/aws/AwsSigv4Signer.js#L73

Reply:
aren't we using our latest js client?, let's create an GitHub issue to update it

Reply:
No we are not. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/yarn.lock#L2341-L2344